### PR TITLE
Don't involve evloop after fork in System::Process.spawn (UNIX)

### PR DESCRIPTION
**Pull request by ysbaddaden**
Refactors spawning child processes on UNIX that relies on fork/exec to not involve the event loop after fork and before exec. We still continue to rely on the eventloop in the parent process, of course.

Extracted from #14959 that exhibit issues with fork/exec.

### CODE REVIEW

File: src/kernel.cr
Comment (ysbaddaden) [inline src/kernel.cr:1]: This change should be in its own PR: it's related to fork, not fork/exec.

Comment (ysbaddaden) [reply src/kernel.cr:1]: -> #14975

File: src/crystal/system/unix/process.cr [hunk 1]
```diff
@@ -231,44 +231,47 @@ struct Crystal::System::Process
...
+        FileDescriptor.write_fully(w, pointerof(errno), 4)
       rescue ex
-        writer_pipe.write_byte(0)
+        byte = 0_u8
         message = ex.inspect_with_backtrace
-        writer_pipe.write_bytes(message.bytesize)
-        writer_pipe << message
-        writer_pipe.close
+        FileDescriptor.write_fully(w, pointerof(byte), 1)
+        FileDescriptor.write_fully(w, message.to_unsafe, message.bytesize)
```
Comment (straight-shoota) [inline src/crystal/system/unix/process.cr:249]: **polish:**
```suggestion
        FileDescriptor.write_fully(w, message.to_slice)
```

File: src/crystal/system/unix/file_descriptor.cr [hunk 2]
```diff
@@ -252,6 +252,19 @@ module Crystal::System::FileDescriptor
     io
   end
 
+  def self.write_fully(fd : LibC::Int, ptr : Pointer, size : Int32)
+    write_fully Slice.new(ptr.as(UInt8*), size)
+  end
```
Comment (straight-shoota) [inline src/crystal/system/unix/file_descriptor.cr]: **polish:** I think this signature is a bit misleading about the pointer and size parameters. The the size refers to the bytesize of the pointer value, not the number of values in that pointer (as in `Slice.new(pointer : Pointer(T), size : Int)`).

And we actually don't need `size` here because it can be derived from `sizeof(ptr.value)`.
So maybe we could simplify the def like that
```suggestion
  def self.write_fully(fd : LibC::Int, ptr : Pointer, size : Int32 = 1)
    write_fully Slice.new(ptr, size).unsafe_slice_of(UInt8)
  end
```

Further down we can skip the size argument except for `message.bytesize`, although that usage could pass `message.to_slice` directly, so maybe the `size` parameter here is unnecessary.

**Review by straight-shoota — APPROVED** [review]
